superduperdb/vector_search/in_memory.py

```diff
@@ -71,6 +73,21 @@ def find_nearest_from_array(self, h, n=100, within_ids=None):
         return _ids, scores
 
     def add(self, items: t.Sequence[VectorItem]) -> None:
+        if len(self._cache) < self._CACHE_SIZE:
+            for item in items:
+                self._cache.append(item)
+        else:
+            self._add(self._cache)
+            self._cache = []
+
+    def post_create(self):
+        self._add(self._cache)
+        self._cache = []
+
+    def _add(self, items: t.Sequence[VectorItem]) -> None:
+        if len(self._cache) < 10000:
+            self._cache.append(items)
+
```

Comment:
Why do we need it?

Reply:
This is needed since we were doing np transactions under a for loop which causes side effects

with the above change we are accumulating 10000 vectors and then adding them to inmemory in one go.